### ISSUE TITLE
Bubbletree error R packages missing

### ISSUE THREAD

**Issue by choosehappy** [chapmanb/bcbio-nextgen#2180]
Follow up to #1229 

Bubbletree directory is now being created with these contents:

```
root@51853b033dc8:/data/CNV.valid/P1/work/heterogeneity/T1/bubbletree# ls -l
total 1284
-rw-r--r-- 1 root root     147 Dec 12 23:50 T1-cnvkit-prep.csv
-rw-r--r-- 1 root root    2576 Dec 12 23:50 b1-vepeffects-annotated-vardict-prep-run.R
-rw-r--r-- 1 root root 1306453 Dec 12 23:50 b1-vepeffects-annotated-vardict-prep.csv
root@51853b033dc8:/data/CNV.valid/P1/work/heterogeneity/T1/bubbletree#

```

but some packages appear to be missing:

```
[2017-12-12T23:50Z] Error: package or namespace load failed for 'GenomeInfoDb' in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]):
[2017-12-12T23:50Z]  there is no package called 'GenomeInfoDbData'
[2017-12-12T23:50Z] Error: package 'GenomeInfoDb' could not be loaded
[2017-12-12T23:50Z] Execution halted
[2017-12-12T23:50Z] Uncaught exception occurred
Traceback (most recent call last):
  File "/usr/local/share/bcbio-nextgen/anaconda/lib/python2.7/site-packages/bcbio/provenance/do.py", line 23, in run
    _do_run(cmd, checks, log_stdout, env=env)
  File "/usr/local/share/bcbio-nextgen/anaconda/lib/python2.7/site-packages/bcbio/provenance/do.py", line 103, in _do_run
    raise subprocess.CalledProcessError(exitcode, error_msg)
CalledProcessError: Command 'set -o pipefail; unset R_HOME && unset R_LIBS && export PATH=/usr/local/share/bcbio-nextgen/anaconda/bin:$PATH && /usr/local/share/bcbio-nextgen/anaconda/bin/Rscript /data/CNV.valid/P1/work/heterogeneity/T1/b
ubbletree/b1-vepeffects-annotated-vardict-prep-run.R


```

**Comment by choosehappy** [comment]
I installed it manually like so:

`
set -o pipefail; unset R_HOME && unset R_LIBS && export PATH=/usr/local/share/bcbio-nextgen/anaconda/bin:$PATH && /usr/local/share/bcbio-nextgen/anaconda/bin/R
`
and ran the following in R:

```
source("https://bioconductor.org/biocLite.R")
biocLite("GenomeInfoDbData")
```

and then re-ran the pipeline, which resulted in this (new) error:


```
[2017-12-13T09:37Z] The following objects are masked from 'package:plyr':
[2017-12-13T09:37Z]     arrange, count, desc, failwith, id, mutate, rename, summarise,
[2017-12-13T09:37Z]     summarize
[2017-12-13T09:37Z] The following objects are masked from 'package:GenomicRanges':
[2017-12-13T09:37Z]     intersect, setdiff, union
[2017-12-13T09:37Z] The following object is masked from 'package:GenomeInfoDb':
[2017-12-13T09:37Z]     intersect
[2017-12-13T09:37Z] The following objects are masked from 'package:IRanges':
[2017-12-13T09:37Z]     collapse, desc, intersect, setdiff, slice, union
[2017-12-13T09:37Z] The following objects are masked from 'package:S4Vectors':
[2017-12-13T09:37Z]     first, intersect, rename, setdiff, setequal, union
[2017-12-13T09:37Z] The following objects are masked from 'package:BiocGenerics':
[2017-12-13T09:37Z]     combine, intersect, setdiff, union
[2017-12-13T09:37Z] The following objects are masked from 'package:stats':
[2017-12-13T09:37Z]     filter, lag
[2017-12-13T09:37Z] The following objects are masked from 'package:base':
[2017-12-13T09:37Z]     intersect, setdiff, setequal, union
[2017-12-13T09:37Z] Loading required package: magrittr
[2017-12-13T09:37Z] Warning messages:
[2017-12-13T09:37Z] 1: Installed Rcpp (0.12.13) different from Rcpp used to build dplyr (0.12.10).
[2017-12-13T09:37Z] Please reinstall dplyr to avoid random crashes or undefined behavior.
[2017-12-13T09:37Z] 2: no function found corresponding to methods exports from 'DelayedArray' for: 'acbind', 'arbind'
[2017-12-13T09:37Z] 3: no function found corresponding to methods exports from 'SummarizedExperiment' for: 'acbind', 'arbind'
[2017-12-13T09:37Z] 4: multiple methods tables found for 'rglist'
[2017-12-13T09:37Z] 5: multiple methods tables found for 'Ontology'
[2017-12-13T09:37Z] 6: replacing previous import 'AnnotationDbi::Ontology' by 'BiocGenerics::Ontology' when loading 'GenomicFeatures'
[2017-12-13T09:37Z] Error in .normargSEW0(start, "start") :
[2017-12-13T09:37Z]   'start' must be a numeric vector (or NULL)
[2017-12-13T09:37Z] Calls: GRanges ... new_GRanges -> IRanges -> solveUserSEW0 -> .normargSEW0
[2017-12-13T09:37Z] Execution halted
[2017-12-13T09:37Z] Uncaught exception occurred
Traceback (most recent call last):
  File "/usr/local/share/bcbio-nextgen/anaconda/lib/python2.7/site-packages/bcbio/provenance/do.py", line 23, in run
    _do_run(cmd, checks, log_stdout, env=env)
  File "/usr/local/share/bcbio-nextgen/anaconda/lib/python2.7/site-packages/bcbio/provenance/do.py", line 103, in _do_run
    raise subprocess.CalledProcessError(exitcode, error_msg)
CalledProcessError: Command 'set -o pipefail; unset R_HOME && unset R_LIBS && export PATH=/usr/local/share/bcbio-nextgen/anaconda/bin:$PATH && /usr/local/share/bcbio-nextgen/anaconda/bin/Rscript /data/CNV.valid/P1/work/heterogeneity/T1/bubbletree/b1-vepeffects-annotated-vardict-prep-run.R
```

I've put the entire contents of the bubbletree directory online for your review here:

https://www.dropbox.com/sh/ek23vmjjphrsui0/AADrHfKeG3rTSmoH7QUFmNXra?dl=0

**Comment by chapmanb** [comment]
Thank you for this report and for your patience. Apologies for missing this earlier. I've pushed fixes which resolve both of these issues: the conda package now depends on bioconductor-genomeinfodbdata and the bcbio code more flexibly handles input CNVkit files. In this case the `T1-cnvkit-prep.csv` file from CNVkit is problematic because the input cns file was not in the expected order. I've fixed this by not requiring an ordering of columns and instead using the header. If you remove that file from your bubbletree directory and re-run in place it should hopefully do the right thing now and generate a correct file.

Thank you again for the report and hope this gets your analysis finished.
